### ISSUE TITLE
MaskFormer baseline

### ISSUE THREAD

**Issue by CJcool06** [ChrisFuscoMasters/TransformerImpl#19]
Running a long experiment will test the validity of the current implementation and give us a standardised experimentation procedure.

**Pre-requisites:**
- [x] #18
- [x] Read [TAO-OW](https://arxiv.org/abs/2104.11221)

**Project Plan:**
1. Choose evaluation metric: 
**Proposal Recall (background & objectness scores)**
2. Decide if we are using pre-trained models:
**Yes**
3. If not, choose training dataset: 
**None**
4. Choose evaluation dataset:
**[TAO-OW](https://arxiv.org/abs/2104.11221) (Validation set)**
5. Choose evaluation models:
**[MaskFormer](https://github.com/facebookresearch/MaskFormer/blob/main/MODEL_ZOO.md#coco-panoptic-segmentation), [Mask R-CNN](https://github.com/facebookresearch/detectron2/blob/main/MODEL_ZOO.md#coco-panoptic-segmentation-baselines-with-panoptic-fpn), [Mask2Former](https://github.com/facebookresearch/Mask2Former/blob/main/MODEL_ZOO.md#coco-model-zoo)**

**Experimental Plan:**
1. Download and unpack TAO-OW dataset
2. Download pre-trained Mask R-CNN and MaskFormer models
3. Calculate mIOU and recall metrics for predictions
4. Generate results

**What's been done:**
- Re-written codebase for easier use throughout project lifespan
- Added a post-processing step to get bounding boxes from panoptic segmentation masks
- Simplified model configs
- New config options
- Simplified running by using `bash` scripts and `make`
- Inference and metric calculation happens at once
- Reduced amount of files resulting from inference (hundreds instead of hundreds of thousands)
- Easier visualisation of masks and bounding boxes

**Comment by CJcool06** [comment]
Notes:
- The box annotations in TAO are in the format of `[x, y, w, h]`, w: width, h: height.

**Comment by CJcool06** [comment]
This issue is being closed for organisational reasons. The work outlined in this issue has been completed, however the overall project is being continued in a different issue.
